docs/howto/use-with-nfs.md

```diff
@@ -0,0 +1,292 @@
+   ```
+   Follow the prompts to set up the Kerberos realm.
+
+2. **Add Principals:**
+   In Kerberos, a principal is a unique identity used for authentication.
+
+   - Add a principal for the NFS server:
+     This principal is used by the NFS client to authenticate when mounting an NFS directory.
+     ```bash
+     sudo kadmin.local addprinc -randkey nfs/server.example.com
+     ```
+
+   - Add a principal for the user `alice`:
+     This principal is used for authentication when the user accesses the mounted NFS directory.
+     ```bash
+     sudo kadmin.local addprinc alice
+     ```
+     Set a password for the user `alice`.
+
+3. **Generate Keytabs:**
+
+   A *keytab* is a file that contains Kerberos principals and their associated secret keys. It allows services (such as NFS) to authenticate without needing to input a password each time.
+
+   - Export the keytab for the NFS server and the user `alice`:
+     ```bash
+     sudo kadmin.local ktadd -k /etc/krb5.keytab nfs/server.example.com
+     ```
+
+---
+
+#### Step 3: Configure NFS Server
+
+1. **Create and Configure the Shared Directory:**
+
+   You’ll need to create the directory to share via NFS and configure the `/etc/exports` file to define it.
+
+   - **Create a directory owned by `alice`:**
+     ```bash
+     sudo mkdir -p /srv/nfs/shared/alice
+     sudo chown alice:alice /srv/nfs/shared/alice
+     ```
+
+   - **Configure exports:**
+     Edit the `/etc/exports` file to define the shared directory:
+     ```bash
+     sudo editor /etc/exports
+     ```
+     Add a line like this:
+     ```
+     /srv/nfs/shared *(rw,sync,no_subtree_check,sec=krb5)
+     ```
+
+2. **Configure IDMAP:**
+   Edit the IDMAP configuration:
+   ```bash
+   sudo editor /etc/idmapd.conf
+   ```
+   Ensure the following is set:
+   ```
+   [General]
+   Domain = example.com
+   ```
+
+3. **Restart Services:**
+   Restart the required services:
+   ```bash
+   sudo systemctl restart nfs-kernel-server rpcbind rpc-svcgssd
+   ```
+
+4. **Verify Running Services:**
+   Check the status of the relevant services:
+   ```bash
+   sudo systemctl status nfs-kernel-server rpcbind rpc-svcgssd
+   ```
+
+---
+
+### Steps for the Client
+
+#### Step 1: Install Required Packages
+
+1. **Install Packages:**
+   On the NFS client, run:
+   ```bash
+   sudo apt install -y nfs-common krb5-user rpcbind
+   ```
+
+2. **Handle Kerberos Configuration Prompts:**
+   During the installation of `krb5-user`, you will be prompted to provide configuration details for Kerberos again. Enter the same details as before:
+
+   - **Default Kerberos version 5 realm:**
+     ```
+     EXAMPLE.COM
+     ```
+
+   - **Kerberos servers for your realm:**
+     ```
+     server.example.com
+     ```
+
+   - **Administrative server for your Kerberos realm:**
+     ```
+     server.example.com
+     ```
+
+---
+
+#### Step 2: Copy the Kerberos Keytab file
+
+1. **Copy Keytab File:**
+   Securely copy the keytab from the server to the client:
```

Comment:
in the commands, the keytab is being copied and access permissions are also being modified -- consider stating the latter also.

Reply:
done